src/graphql/utilities/type_info.py

```diff
@@ -136,15 +137,25 @@ def get_enum_value(self) -> Optional[GraphQLEnumValue]:
         return self._enum_value
 
     def enter(self, node: Node) -> None:
-        method = getattr(self, "enter_" + node.kind, None)
+        method = self._get_method("enter", node.kind)
         if method:
             method(node)
 
     def leave(self, node: Node) -> None:
-        method = getattr(self, "leave_" + node.kind, None)
+        method = self._get_method("leave", node.kind)
         if method:
             method()
 
+    def _get_method(self, direction: str, kind: str) -> Optional[Callable[[], None]]:
+        key = (direction, kind)
+        if key in self._visit_fns:
+            return self._visit_fns[key]
+
+        fn = getattr(self, f"{direction}_{kind}", None)
+        self._visit_fns[key] = fn
+        return fn
```

Comment:
I think there is a cleaner way to do this:

```suggestion
        if key not in self._visit_fns:
            fn = getattr(self, f"{direction}_{kind}", None)
            self._visit_fns[key] = fn
        return self._visit_fns[key]
```

Reply:
LGTM, see remark regarding visitor.py